src/components/tag-filter/__tests__/tag-filter.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { render, screen, userEvent } from '@/test-utils/rtl';
+
+import TagFilter from '../tag-filter';
+import { type TagFilterOptionConfig } from '../tag-filter.types';
+
+jest.mock('../selectable-tag/selectable-tag', () =>
+  jest.fn(({ value, onClick, children, startEnhancer }) => (
+    <button
+      data-testid={`selectable-tag-${children}`}
+      data-selected={value}
+      onClick={onClick}
+    >
+      {startEnhancer && startEnhancer()}
+      {children}
+    </button>
+  ))
+);
+
+const MOCK_OPTIONS_CONFIG = {
+  opt1: {
+    label: 'Option 1',
+    startEnhancer: () => <div data-testid="enhancer-opt1" />,
+  },
+  opt2: {
+    label: 'Option 2',
+  },
+  opt3: {
+    label: 'Option 3',
+    startEnhancer: () => <div data-testid="enhancer-opt3" />,
+  },
+} as const satisfies Record<string, TagFilterOptionConfig>;
+
+type MockOption = keyof typeof MOCK_OPTIONS_CONFIG;
+
+describe(TagFilter.name, () => {
+  it('renders without errors', () => {
+    setup({});
+
+    expect(screen.getByText('Mock Label')).toBeInTheDocument();
+    expect(screen.getByText('Show all')).toBeInTheDocument();
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
+    expect(screen.getByText('Option 2')).toBeInTheDocument();
+    expect(screen.getByText('Option 3')).toBeInTheDocument();
+  });
+
+  it('renders all tags from optionsConfig', () => {
+    setup({});
+
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
+    expect(screen.getByText('Option 2')).toBeInTheDocument();
+    expect(screen.getByText('Option 3')).toBeInTheDocument();
+    expect(screen.getByTestId('enhancer-opt1')).toBeInTheDocument();
+    expect(screen.queryByTestId('enhancer-opt2')).not.toBeInTheDocument();
+    expect(screen.getByTestId('enhancer-opt3')).toBeInTheDocument();
+  });
+
+  it('renders "Show all" tag', () => {
+    setup({});
+
+    expect(screen.getByText('Show all')).toBeInTheDocument();
+  });
+
+  it('calls onChangeValues when clicking an individual tag to select it', async () => {
+    const { user, mockOnChangeValues } = setup({
+      values: [],
+    });
+
+    const option1Tag = screen.getByText('Option 1');
+    await user.click(option1Tag);
+
+    expect(mockOnChangeValues).toHaveBeenCalledWith(['opt1']);
+  });
+
+  it('calls onChangeValues when clicking an individual tag to deselect it', async () => {
+    const { user, mockOnChangeValues } = setup({
+      values: ['opt1', 'opt2'],
+    });
+
+    const option1Tag = screen.getByText('Option 1');
+    await user.click(option1Tag);
+
+    expect(mockOnChangeValues).toHaveBeenCalledWith(['opt2']);
+  });
+
+  it('calls onChangeValues with all tag keys when clicking "Show all" and no values are selected', async () => {
```

Comment:
Can you add an overview of user interactions in the Test Plan? 

I'm wondering why "all tag keys" are provided when "no values are selected"? I wonder if there's a typo in the latter?